libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1458,18 +1524,24 @@ string YulUtilFunctions::readFromCalldata(Type const& _type)
 	return readFromMemoryOrCalldata(_type, true);
 }
 
-string YulUtilFunctions::updateStorageValueFunction(Type const& _type, std::optional<unsigned> const& _offset)
+string YulUtilFunctions::updateStorageValueFunction(
+	Type const& _toType,
+	Type const* _fromType,
+	std::optional<unsigned> const& _offset
+)
 {
 	string const functionName =
 		"update_storage_value_" +
 		(_offset.has_value() ? ("offset_" + to_string(*_offset)) : "") +
-		_type.identifier();
+		(_fromType ? "_from_" + _fromType->identifier() : "") +
+		"_to_" +
+		_toType.identifier();
 
 	return m_functionCollector.createFunction(functionName, [&] {
-		if (_type.isValueType())
+		if (_toType.isValueType())
 		{
```

Comment:
See also my comment about this function in the header, so maybe we'll want to change this generally... but here we should probably assert ``solAssert(!_fromType || *_fromType == _toType, "");`` or something like that?